### ISSUE TITLE
cerebras llm - 422 - property metadata is unsupported

### ISSUE THREAD

**Issue by sam-trost** [livekit/agents#1274]
The openai llm component is including a `metadata` and `store` fields that are not supported by all openai compliant providers (ex. cerebras, groq)

```bash
raise self._make_status_error_from_response(err.response) from None
openai.UnprocessableEntityError: Error code: 422 - {'message': "metadata: property 'metadata' is unsupported", 'type': 'invalid_request_error', 'param': 'validation_error', 'code': 'wrong_api_format'}
```

Example agent

```python
import os
import asyncio

from livekit.agents import JobContext, WorkerOptions, cli, JobProcess
from livekit.agents.llm import (
    ChatContext,
    ChatMessage,
)
from livekit.agents.voice_assistant import VoiceAssistant
from livekit.plugins import deepgram, silero, elevenlabs, openai

from dotenv import load_dotenv

load_dotenv()


def prewarm(proc: JobProcess):
    proc.userdata["vad"] = silero.VAD.load()


async def entrypoint(ctx: JobContext):
    initial_ctx = ChatContext(
        messages=[
            ChatMessage(
                role="system",
                content="You are a voice assistant. Pretend we're having a human conversation, no special formatting or headings, just natural speech.",
            )
        ]
    )

    assistant = VoiceAssistant(
        vad=ctx.proc.userdata["vad"],
        stt=deepgram.STT(),
        llm=openai.LLM.with_cerebras((
            base_url="https://api.cerebras.ai/v1",
            api_key=os.environ.get("CEREBRAS_API_KEY"),
            model="llama-3.3-70b",
        ),
        tts=elevenlabs.TTS(),
        chat_ctx=initial_ctx,
    )

    await ctx.connect()
    assistant.start(ctx.room)
    await asyncio.sleep(1)
    await assistant.say("Hi there, how are you doing today?", allow_interruptions=True)


if __name__ == "__main__":
    cli.run_app(WorkerOptions(entrypoint_fnc=entrypoint, prewarm_fnc=prewarm))

```

**Comment by SyedAddanACH** [comment]
Having the same issue with groq. Hopefully your PR gets merged soon! ✌🏽

**Comment by robertvy** [comment]
having this issue on azure:

` {"message": "failed to generate LLM completion, retrying in 5.0s\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.12/site-packages/livekit/plugins/openai/llm.py\", line 735, in _run\n    stream = await self._client.chat.completions.create(\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.12/site-packages/openai/resources/chat/completions.py\", line 1720, in create\n    return await self._post(\n           ^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.12/site-packages/openai/_base_client.py\", line 1843, in post\n    return await self.request(cast_to, opts, stream=stream, stream_cls=stream_cls)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.12/site-packages/openai/_base_client.py\", line 1537, in request\n    return await self._request(\n           ^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.12/site-packages/openai/_base_client.py\", line 1638, in _request\n    raise self._make_status_error_from_response(err.response) from None\nopenai.BadRequestError: Error code: 400 - {'error': {'message': \"Unknown parameter: 'store'.\", 'type': 'invalid_request_error', 'param': 'store', 'code': 'unknown_parameter'}}\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.12/site-packages/livekit/agents/llm/llm.py\", line 149, in _main_task\n    return await self._run()\n           ^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.12/site-packages/livekit/plugins/openai/llm.py\", line 771, in _run\n    raise APIStatusError(\nlivekit.agents._exceptions.APIStatusError: Error code: 400 - {'error': {'message': \"Unknown parameter: 'store'.\", 'type': 'invalid_request_error', 'param': 'store', 'code': 'unknown_parameter'}}", "level": "WARNING", "name": "livekit.agents", "llm": "livekit.plugins.openai.llm.LLM", "attempt": 1, "pid": 239, "job_id": "AJ_Mbjmygtrs6wC", "timestamp": "2024-12-22T21:40:25.436302+00:00"}`

**Comment by davidzhao** [comment]
sorry about the breakage.. just released `0.12.4` that fixes this.
